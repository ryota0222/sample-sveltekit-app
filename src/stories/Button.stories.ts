import type { Meta, StoryObj } from '@storybook/svelte';
import Button from './Button.svelte';
import { within, expect } from '@storybook/test';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories
const meta = {
	title: 'Example/Button',
	component: Button,
	tags: ['autodocs'],
	argTypes: {
		backgroundColor: { control: 'color' },
		size: {
			control: { type: 'select' },
			options: ['small', 'medium', 'large']
		}
	}
} satisfies Meta<Button>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Primary: Story = {
	args: {
		primary: true,
		label: 'Button',
		disabled: true
	},
	play: async ({ canvasElement }) => {
		const canvas = within(canvasElement);
		const button = canvas.getByRole('button', { name: /Button/i });
		expect(button.getAttribute('disabled')).toBe('');
	}
};

export const Secondary: Story = {
	args: {
		label: 'Button'
	}
};

export const Large: Story = {
	args: {
		size: 'large',
		label: 'Button'
	}
};

export const Small: Story = {
	args: {
		size: 'small',
		label: 'Button'
	}
};
