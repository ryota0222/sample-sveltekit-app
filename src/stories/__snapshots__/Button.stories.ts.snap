// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/Button Large smoke-test 1`] = `
<button type="button"
        class="storybook-button storybook-button--large storybook-button--secondary"
        style
>
  Button
</button>
`;

exports[`Example/Button Primary play-test 1`] = `
<button type="button"
        class="storybook-button storybook-button--medium storybook-button--primary"
        style
        disabled
>
  Button
</button>
`;

exports[`Example/Button Secondary smoke-test 1`] = `
<button type="button"
        class="storybook-button storybook-button--medium storybook-button--secondary"
        style
>
  Button
</button>
`;

exports[`Example/Button Small smoke-test 1`] = `
<button type="button"
        class="storybook-button storybook-button--small storybook-button--secondary"
        style
>
  Button
</button>
`;
